src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1165,14 +1171,20 @@ FailureContext retryFailed(
                 m_testContext);
         r.setStatus(TestResult.FAILURE);
         result.add(r);
-        runTestListeners(r);
+        runTestResultListener(r);
         m_notifier.addFailedTest(testMethod, r);
       } // catch
     }
 
     return result;
   }
 
+  private void invokeListenersForSkippedTestResult(ITestResult r, InvokedMethod invokedMethod) {
```

Comment:
The param could be testMethod instead of InvokedMethod. Then the build logic will be shared across all usages.

Reply:
`IInvokedMethod` object is required for calling `org.testng.internal.Invoker#runInvokedMethodListeners`